yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala

```diff
@@ -38,24 +38,30 @@ private[spark] class ClientArguments(args: Array[String], sparkConf: SparkConf)
   var amMemory: Int = 512 // MB
   var appName: String = "Spark"
   var priority = 0
+  var isClusterMode = false
```

Comment:
this should be the following instead. Otherwise we might forget to set one but not the other in the future
```
private def isClusterMode: Boolean = userClass != null
```